src/pages/ProfilePage/index.tsx

```diff
@@ -0,0 +1,190 @@
+import { useContext, useState, useEffect } from "react";
+import { AuthContext } from "../../AuthContext";
+import { Button } from "../../components/Button/index";
+import {
+  addDoc,
+  collection,
+  deleteDoc,
+  doc,
+  getDocs,
+  updateDoc,
+} from "firebase/firestore";
+import { db } from "../../firebase";
+import { QuoteCard } from "../../components/QuoteCard";
+import { useQuotesContext } from "../../QuotesContextProvider";
+
+export const ProfilePage = () => {
+  const authContext = useContext(AuthContext);
+  const [quoteInput, setQuoteInput] = useState<string>("");
+  const [authorInput, setAuthorInput] = useState<string>("");
+  const [message, setMessage] = useState<string | null>(null);
+
+  const [editQuoteId, setEditQuoteId] = useState<string | null>(null);
+  const [editQuoteInput, setEditQuoteInput] = useState("");
+  const [editAuthorInput, setEditAuthorInput] = useState("");
+  const [addNewQuote, setaddNewQuote] = useState<boolean>(false);
+
+  const quotes = useQuotesContext();
+  const userEmail = authContext?.user?.email?.toUpperCase().split("@")[0];
+  const userID = authContext?.user?.uid;
+  const isMessagePositive = message?.includes("unsuccessfully") == false;
+
+  useEffect(() => {
+    if (message) {
+      const timeout = setTimeout(() => {
+        setMessage(null);
+      }, 3000);
+
+      return () => clearTimeout(timeout);
+    }
+  }, [message]);
+
+  async function addQuote(e: React.FormEvent) {
+    try {
+      e.preventDefault();
+      await addDoc(collection(db, "quotes"), {
+        author: authorInput,
+        quote: quoteInput,
+        likedBy: 0,
+        whoAddQuote: userID,
+      });
+
+      setMessage("Quote added successfully!");
+      setQuoteInput("");
+      setAuthorInput("");
+    } catch (error) {
+      setMessage("Quote added unsuccessfully!");
+      console.log(error + "Quote added unsuccessfully!");
+    }
+  }
+
+  async function deleteQuote(id: string) {
+    try {
+      await deleteDoc(doc(db, "quotes", id));
```

Comment:
Same here as in the updateQuote function. It's important to check that only the user that created the quote can delete the quote.